### PR TITLE
Tpm2 pcr value post merge

### PR DESCRIPTION
**Pull request by ddstreet**
This contains various commits to address the post-merge comments for #28398

### CODE REVIEW

**Comment by ddstreet** [comment]
the centos test failures seem like infrastructure issues, with lots of

`2023-08-21 23:28:23,723 [_client/_send_single_request] INFO: HTTP Request: POST https://duffy.ci.centos.org/api/v1/sessions "HTTP/1.1 422 Unprocessable Entity"`

**Comment by ddstreet** [comment]
rebased on main

**Comment by ddstreet** [comment]
rebased on #28519, now starting to go over review comments

File: src/shared/tpm2-util.c [hunk 1]
```diff
@@ -1282,22 +1282,15 @@ void tpm2_tpms_pcr_selection_move(TPMS_PCR_SELECTION *a, TPMS_PCR_SELECTION *b)
         tpm2_tpms_pcr_selection_from_mask(0, b->hash, b);
 }
 
+#define FOREACH_TPMS_PCR_SELECTION_IN_TPML_PCR_SELECTION(tpms, tpml)    \
+        _FOREACH_TPMS_PCR_SELECTION_IN_TPML_PCR_SELECTION(tpms, tpml, UNIQ_T(l, UNIQ))
+#define _FOREACH_TPMS_PCR_SELECTION_IN_TPML_PCR_SELECTION(tpms, tpml, l) \
+        for (typeof(tpml) l = (tpml); l; l = NULL)                      \
```
Comment (poettering) [inline src/shared/tpm2-util.c]: enclose l in `()`

File: src/shared/tpm2-util.c [hunk 2]
```diff
@@ -3507,36 +3507,42 @@ static int tpm2_build_sealing_policy(
...
+static const struct {
+        TPM2_ECC_CURVE tpm2_ecc_curve_id;
+        int openssl_ecc_curve_id;
+} TPM2_OPENSSL_ECC_CURVE_TABLE[] = {
+        { TPM2_ECC_NIST_P192, NID_X9_62_prime192v1, },
+        { TPM2_ECC_NIST_P224, NID_secp224r1, },
+        { TPM2_ECC_NIST_P256, NID_X9_62_prime256v1, },
+        { TPM2_ECC_NIST_P384, NID_secp384r1, },
+        { TPM2_ECC_NIST_P521, NID_secp521r1, },
+        { TPM2_ECC_SM2_P256,  NID_sm2, },
```
Comment (poettering) [inline src/shared/tpm2-util.c]: doesn't matter, but we'd usually also align the closing `}` on the same column

File: src/shared/tpm2-util.c [hunk 3]
```diff
@@ -3507,36 +3507,42 @@ static int tpm2_build_sealing_policy(
 }
 
 #if HAVE_OPENSSL
-static int tpm2_ecc_curve_from_openssl_curve_id(int curve_id, TPM2_ECC_CURVE *ret) {
+static const struct {
+        TPM2_ECC_CURVE tpm2_ecc_curve_id;
+        int openssl_ecc_curve_id;
+} TPM2_OPENSSL_ECC_CURVE_TABLE[] = {
```
Comment (poettering) [inline src/shared/tpm2-util.c]: why uppercase? usually we don't uppercase our variable names

**Comment by poettering** [comment]
lgtm

File: src/shared/tpm2-util.c (hunk 1 above)
Comment (ddstreet) [reply src/shared/tpm2-util.c]: ack

File: src/shared/tpm2-util.c (hunk 3 above)
Comment (ddstreet) [reply src/shared/tpm2-util.c]: ack, updated

File: src/shared/tpm2-util.c (hunk 2 above)
Comment (ddstreet) [reply src/shared/tpm2-util.c]: will do

**Comment by bluca** [comment]
ubuntu CI didn't run for some reason, triggered a couple of jobs

**Comment by bluca** [comment]
@yuwata this looks new (but unrelated to this PR):

```
772s  573/1183 systemd:network / test-dhcp-client                               FAIL              2.31s   killed by signal 6 SIGABRT
772s >>> PATH=/tmp/autopkgtest.mEVoY1/build.Hh9/systemd/build-deb:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin PROJECT_BUILD_ROOT=/tmp/autopkgtest.mEVoY1/build.Hh9/systemd/build-deb MALLOC_PERTURB_=243 SYSTEMD_LANGUAGE_FALLBACK_MAP=/tmp/autopkgtest.mEVoY1/build.Hh9/systemd/src/locale/language-fallback-map /tmp/autopkgtest.mEVoY1/build.Hh9/systemd/build-deb/test-dhcp-client
772s ――――――――――――――――――――――――――――――――――――― ✀  ―――――――――――――――――――――――――――――――――――――
772s stderr:
772s Assertion 'client' failed at src/libsystemd-network/sd-dhcp-client.c:252, function sd_dhcp_client_set_request_option(). Ignoring.
772s Assertion 'client' failed at src/libsystemd-network/sd-dhcp-client.c:275, function sd_dhcp_client_set_request_address(). Ignoring.
772s Assertion 'client' failed at src/libsystemd-network/sd-dhcp-client.c:287, function sd_dhcp_client_set_ifindex(). Ignoring.
772s Assertion 'ifindex > 0' failed at src/libsystemd-network/sd-dhcp-client.c:289, function sd_dhcp_client_set_ifindex(). Ignoring.
772s Assertion 'ifindex > 0' failed at src/libsystemd-network/sd-dhcp-client.c:289, function sd_dhcp_client_set_ifindex(). Ignoring.
772s Assertion 'ifindex > 0' failed at src/libsystemd-network/sd-dhcp-client.c:289, function sd_dhcp_client_set_ifindex(). Ignoring.
772s lo: DHCPv4 client: FREE
772s DHCPv4 client: FREE
772s DHCPv4 client: STARTED on ifindex 42
772s Assertion '!sd_dhcp_client_is_running(client)' failed at src/libsystemd-network/sd-dhcp-client.c:489, function sd_dhcp_client_set_iaid_duid_en(). Ignoring.
772s DHCPv4 client: STOPPED
772s DHCPv4 client: FREE
772s DHCPv4 client: STARTED on ifindex 42
772s Assertion '!sd_dhcp_client_is_running(client)' failed at src/libsystemd-network/sd-dhcp-client.c:489, function sd_dhcp_client_set_iaid_duid_en(). Ignoring.
772s Assertion 'sd_event_loop(e) >= 0' failed at src/libsystemd-network/test-dhcp-client.c:525, function test_addr_acq(). Aborting.

```

**Pull request merged**